packages/backend-common/src/remoteConfig.ts

```diff
@@ -0,0 +1,176 @@
+    for (const subscriber of this.subscribers) {
+      try {
+        subscriber();
+      } catch (error) {
+        this.logger.error(`Config subscriber threw error, ${error}`);
+      }
+    }
+  }
+
+  subscribe(onChange: () => void): { unsubscribe: () => void } {
+    this.subscribers.push(onChange);
+    return {
+      unsubscribe: () => {
+        const index = this.subscribers.indexOf(onChange);
+        if (index >= 0) {
+          this.subscribers.splice(index, 1);
+        }
+      },
+    };
+  }
+
+  has(key: string): boolean {
+    return this.config.has(key);
+  }
+
+  keys(): string[] {
+    return this.config.keys();
+  }
+
+  get<T = JsonValue>(key?: string): T {
+    return this.config.get(key);
+  }
+
+  getOptional<T = JsonValue>(key?: string): T | undefined {
+    return this.config.getOptional(key);
+  }
+
+  getConfig(key: string): Config {
+    return this.config.getConfig(key);
+  }
+
+  getOptionalConfig(key: string): Config | undefined {
+    return this.config.getOptionalConfig(key);
+  }
+
+  getConfigArray(key: string): Config[] {
+    return this.config.getConfigArray(key);
+  }
+
+  getOptionalConfigArray(key: string): Config[] | undefined {
+    return this.config.getOptionalConfigArray(key);
+  }
+
+  getNumber(key: string): number {
+    return this.config.getNumber(key);
+  }
+
+  getOptionalNumber(key: string): number | undefined {
+    return this.config.getOptionalNumber(key);
+  }
+
+  getBoolean(key: string): boolean {
+    return this.config.getBoolean(key);
+  }
+
+  getOptionalBoolean(key: string): boolean | undefined {
+    return this.config.getOptionalBoolean(key);
+  }
+
+  getString(key: string): string {
+    return this.config.getString(key);
+  }
+
+  getOptionalString(key: string): string | undefined {
+    return this.config.getOptionalString(key);
+  }
+
+  getStringArray(key: string): string[] {
+    return this.config.getStringArray(key);
+  }
+
+  getOptionalStringArray(key: string): string[] | undefined {
+    return this.config.getOptionalStringArray(key);
+  }
+}
+
+// A global used to ensure that only a single file watcher is active at a time.
+let currentCancelFunc: () => void;
+
+/**
+ * Load configuration for a Backend.
+ *
+ * This function should only be called once, during the initialization of the backend.
+ *
+ * @public
+ */
+export async function loadRemoteBackendConfig(options: {
+  logger: Logger;
+  // process.argv or any other overrides
+  argv: string[];
+}): Promise<Config> {
+  const args = parseArgs(options.argv);
+  const configPaths: string[] = [args.config ?? []].flat();
+  const configUrls: string[] = [args.config ?? []].flat();
+
+  const config = new RemoteObservableConfigProxy(options.logger);
+
+  /* eslint-disable-next-line no-restricted-syntax */
+  const paths = findPaths(__dirname);
+
+  const configs = await loadConfig({
+    configRoot: paths.targetRoot,
+    configPaths: configPaths,
+    configUrls: configUrls,
+    watch: {
```

Comment:
I know this does not makes sense, but I was not sure I understood how this observable patter work, tried replacing it with Rx but I thought it would be over kill for now